packages/apps/plugins/plugin-sketch/src/components/Sketch.tsx

```diff
@@ -30,83 +25,116 @@ export type SketchComponentProps = {
-
-  const zoomToContent = (animate = true) => {
-    const commonBounds = editor?.allShapesCommonBounds;
-    if (editor && width && height && commonBounds?.width && commonBounds?.height) {
-      const padding = 40;
-      // NOTE: Objects culled (unstyled) if outside of bounds.
-      const zoom = Math.min(maxZoom, (width - padding) / commonBounds.width, (height - padding) / commonBounds.height);
-      if (zoom <= 0) {
-        throw new Error('Sketch has non-positive size.');
-      }
-      const center = {
-        x: (width - commonBounds.width * zoom) / 2 / zoom - commonBounds.minX,
-        y: (height - commonBounds.height * zoom) / 2 / zoom - commonBounds.minY,
-      };
-      editor.animateCamera(center.x, center.y, zoom, animate ? { duration: 250 } : undefined);
-      setReady(true);
+  useEffect(() => {
+    if (!editor || !adapter || width === undefined || height === undefined) {
+      return;
     }
-  };
 
-  useEffect(() => {
-    if (!autoZoom || width < 1 || (height ?? 0) < 1) {
+    editor.zoomToFit({ duration: 0 });
+    editor.resetZoom();
+
+    setReady(true);
+    if (!autoZoom) {
       return;
     }
 
-    editor?.updateViewportScreenBounds();
-    zoomToContent(false);
+    const zoom = (animate = true) => {
+      zoomToContent(editor, width, height, maxZoom, animate);
+    };
 
-    const onUpdate = debounce(zoomToContent, 200);
-    const subscription = store.listen(() => onUpdate(true), { scope: 'document' });
-    return () => subscription();
-  }, [editor, width, height]);
+    zoom(false);
+    const onUpdate = debounce(zoom, 200);
+    const subscription = readonly ? adapter.store!.listen(() => onUpdate(true), { scope: 'document' }) : undefined;
+    return () => subscription?.();
+  }, [editor, adapter, width, height, autoZoom]);
 
-  // https://github.com/tldraw/tldraw/blob/main/packages/ui/src/lib/TldrawUi.tsx
-  // TODO(burdon): Customize assets: https://tldraw.dev/docs/assets
+  if (!adapter) {
+    return null;
+  }
+
+  // https://tldraw.dev/docs/user-interface
   return (
     <div
       ref={containerRef}
       style={{ visibility: ready ? 'visible' : 'hidden' }}
-      className={mx(
-        'is-full bs-full',
-        // 14Override z-index.
-        '[&>div>span>div]:z-0',
-        // 14Hide .tlui-menu-zone
-        '[&>div>main>div:nth-child(1)>div:nth-child(1)]:hidden',
-        // 14Hide .tlui-navigation-zone
-        '[&>div>main>div:nth-child(2)>div:nth-child(1)>div:nth-child(1)]:hidden',
-        // 14Hide .tlui-help-menu
-        '[&>div>main>div:nth-child(2)>div:nth-child(1)>div:nth-child(3)]:hidden',
-        // 14Hide .tlui-debug-panel
-        '[&>div>main>div:nth-child(2)>div:nth-child(2)]:hidden',
-        styles.background,
-        className,
-      )}
+      className={mx('is-full bs-full', className)}
+      onPointerEnter={() => {
+        setActive(!readonly && !adapter.readonly);
+      }}
+      onPointerLeave={() => {
+        setActive(false);
+      }}
```

Comment:
This looks like maybe it was added to make Sketch work better in Stacks/Collections, but it's really awkward when Sketch is presented in the `main` surface.